### PR TITLE
Add List and Details Page for Volume Snapshot Classes

### PR DESCRIPTION
**Pull request by bipuladh**
Screenshot from 2020-07-14 18-08-05
Screenshot from 2020-07-14 18-06-55

### CODE REVIEW

File: frontend/packages/console-app/src/components/volume-snapshot/volume-snapshot-class-details.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  navFactory,
+  Kebab,
+} from '@console/internal/components/utils';
+import { VolumeSnapshotClassKind } from '@console/internal/module/k8s';
+import { DetailsPage } from '@console/internal/components/factory';
+import { volumeSnapshotStatus } from '../../status';
+
+const { editYaml } = navFactory;
+
+const Details: React.FC<DetailsProps> = ({ obj }) => {
```
Comment (gnehapk) [inline frontend/packages/console-app/src/components/volume-snapshot/volume-snapshot-class-details.tsx:14]: nit: you can directly return the `<div className="co-m-pane__body">` here

File: frontend/packages/console-app/src/components/volume-snapshot/volume-snapshot-class.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+type RowProps = {
+  obj: VolumeSnapshotClassKind;
+  index: number;
+  style: object;
+  key: string;
+};
+
+const Row: React.FC<RowProps> = ({ obj, index, style, key }) => {
+  const { name } = obj?.metadata || {};
```
Comment (gnehapk) [inline frontend/packages/console-app/src/components/volume-snapshot/volume-snapshot-class.tsx:46]: `obj` -> `volumeSnapshotClass` something like this

Comment (bipuladh) [reply frontend/packages/console-app/src/components/volume-snapshot/volume-snapshot-class.tsx:48]: Is it really required?

File: frontend/packages/console-app/src/components/volume-snapshot/volume-snapshot-class-details.tsx (hunk 1 above)
Comment (bipuladh) [reply frontend/packages/console-app/src/components/volume-snapshot/volume-snapshot-class-details.tsx]: ack

File: frontend/packages/console-app/src/components/volume-snapshot/volume-snapshot-class.tsx (hunk 2 above)
Comment (gnehapk) [reply frontend/packages/console-app/src/components/volume-snapshot/volume-snapshot-class.tsx:48]: its more understandable IMO.

**Comment by gnehapk** [comment]
@bipuladh Please rebase your PR and the screenshots for the respective pages.

**Comment by bipuladh** [comment]
/assign @rawagner 
cc @spadgett

**Pull request merged**